controllers/ingresscluster/tls.go

```diff
@@ -0,0 +1,43 @@
+package ingresscluster
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"time"
+)
+
+func verifyCertificate(customCert []byte) error {
+	block, _ := pem.Decode(customCert)
+	if block == nil {
+		return fmt.Errorf("failed to decode certificate PEM")
+	}
+	certificate, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		return err
+	}
+	now := time.Now()
+	if now.After(certificate.NotAfter) {
+		return fmt.Errorf("custom TLS certificate is expired")
+	}
+	if now.Before(certificate.NotBefore) {
+		return fmt.Errorf("custom TLS certificate is not valid yet")
+	}
+	return nil
+}
+
+func verifyPrivateKey(customKey []byte) error {
+	block, rest := pem.Decode(customKey)
+	fmt.Println(rest)
```

Comment:
maybe you can use a proper logger for this log?

Reply:
removed.